File: js/caesarCipher.js
function ceasarCypher(word, num) {
  let alphabet = "abcdefghijklmnopqrstuvwxyz";
  // let caps = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
  let newWord = "";
  for (let i = 0; i < word.length; i++) {
    let char = word[i];
    let index = alphabet.indexOf(char);
    let newIndex = (num + index) % 26;
  if (alphabet.indexOf(alphabet[newIndex]) > -1) {
       newWord += alphabet[newIndex];
  } 
  // else if (caps.indexOf(caps[newIndex]) > -1) {
  //     newWord += caps[newIndex];
  // } 
  else {
    newWord += char;
  }
  return newWord;
}

console.log("Boy! What a string!", -5)
console.log("Hello zach168! Yes here.", 5)
console.log("Hello Zach168! Yes here.", -5)

File: python/caesar_cipher.py
def caesar_cipher(string, num):
  alphabet = "abcdefghijklmnopqrstuvwxyz"
  caps = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
  new_word = ""

  for i in string:
    if i in alphabet:
      if alphabet.index(i) + num >= 0:
        new_word += alphabet[(alphabet.index(i) + num) % 26]
      else:
        new_word += alphabet[26 + (alphabet.index(i) + num)]

    elif i in caps:
      if caps.index(i) + num >= 0:
        new_word += caps[(caps.index(i) + num) % 26]
      else:
        new_word += caps[26 + (caps.index(i) + num)] 

    else:
      new_word += i

  return new_word

# print(caesar_cipher("hello", 3))